File: CodeSignal/quasifactorial.java
/* https://app.codesignal.com/challenge/53DMcWRBbZMdsF6vm */

public class quasifactorial {

	public static void main(String[] args) {
		
		int n=4;

		System.out.println(Quasifactorial(n)));
	}
	
	public static int Quasifactorial(int n){

		return n==1 ? 1 : n * Quasifactorial(n-1) - 1;
	}
}

File: CodeSignal/quasifactorial.py
# https://app.codesignal.com/challenge/53DMcWRBbZMdsF6vm

def quasifactorial(n):

    if n==1:
        return 1
    else:
        return n * quasifactorial(n-1) - 1

print(quasifactorial(4))
